src/neural/cuda/layers.cc

```diff
@@ -2073,12 +2104,13 @@ AttentionBody<DataType>::AttentionBody(const LegacyWeights& weights,
   }
 
   int num_encoders = weights.encoder.size();
-  float alpha = (float)pow(2.0 * num_encoders, 0.25);
+  float alpha = (float)pow(2.0 * num_encoders, -0.25);
   for (const auto& enc : weights.encoder) {
     EncoderBlock<DataType>* pW = new EncoderBlock<DataType>(
         enc, scratch, encoder_head_count_, embedding_op_size_, alpha,
         smolgen_global_, smolgen_global_size_, max_batch_size,
-        activations_.smolgen_activation, activations_.ffn_activation);
+        activations_.smolgen_activation, activations_.ffn_activation,
+        new_encoding_ ? 1e-3 : 1e-6);
```

Comment:
Should the 1e-6 be scaled by depth for true equivalence with the old code?

Reply:
After doing some experiments, the scaling results in even lower epsilon values, so not worth the effort.